.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,308 @@
+6. [Contribution Standards](#Standards)<br />
+6.1. [Contribution Checklist](#Checklist)<br />
+6.2. [Licensing of Contributions](#Licensing)<br />
+
+<a name="Overview" />
+
+### 1. Overview
+
+Developing cryptocurrencies is an exciting endeavor that touches a wide variety
+of areas such as wire protocols, peer-to-peer networking, databases,
+cryptography, language interpretation (transaction scripts), RPC, and
+websockets.  They also represent a radical shift to the current fiscal system
+and as a result provide an opportunity to help reshape the entire financial
+system.  There are few projects that offer this level of diversity and impact
+all in one code base.
+
+However, as exciting as it is, one must keep in mind that cryptocurrencies
+represent real money and introducing bugs and security vulnerabilities can have
+far more dire consequences than in typical projects where having a small bug is
+minimal by comparison.  In the world of cryptocurrencies, even the smallest bug
+in the wrong area can cost people a significant amount of money.  For this
+reason, bcoin has a formalized and rigorous development process which
+is outlined on this page.
+
+We highly encourage code contributions, however it is imperative that you adhere
+to the guidelines established on this page.
+
+<a name="MinSkillset" />
+
+### 2. Minimum Recommended Skillset
+
+The following list is a set of core competencies that we recommend you possess
+before you really start attempting to contribute code to the project.  These are
+not hard requirements as we will gladly accept code contributions as long as
+they follow the guidelines set forth on this page.  That said, if you don't have
+the following basic qualifications you will likely find it quite difficult to
+contribute.
+
+- A reasonable understanding of bitcoin at a high level (see the
+  [Required Reading](#ReqReading) section for the original white paper)
+- Experience in some type of C-like language
+- An understanding of data structures and their performance implications
+- Familiarity with unit testing
+- Debugging experience
+- Ability to understand not only the area you are making a change in, but also
+  the code your change relies on, and the code which relies on your changed code
+
+Building on top of those core competencies, the recommended skill set largely
+depends on the specific areas you are looking to contribute to.  For example,
+if you wish to contribute to the cryptography code, you should have a good
+understanding of the various aspects involved with cryptography such as the
+security and performance implications.
+
+<a name="ReqReading" />
+
+### 3. Required Reading
+
+- [Original Satoshi Whitepaper](https://bitcoin.org/bitcoin.pdf) - This is the
+  white paper that started it all.  Having a solid foundation to build on will
+  make the code much more comprehensible.
+- [Don't block the Event Loop](https://nodejs.org/en/docs/guides/dont-block-the-event-loop/)
```

Comment:
Maybe for recommended reading we can add "Mastering Bitcoin" and the bcoin.io guides?